### PR TITLE
MSR - Support Cosmetic Options

### PR DESCRIPTION
**Pull request by dyceron**
Adds some color customization options. The following things can be changed:
- Free-Aim color (locked and unlocked)
- Grapple Aim color (locked and unlocked)
- Energy Tank color
- Aeion Bar color
- Ammo HUD color

image

Merge after OSRR has been updated

### CODE REVIEW

**Review by ThanatosGit — CHANGES_REQUESTED** [review]
Mypy is complaining at multiple lines.
Would also be nice to raise the coverage for the  `cosmetic_patches_dialog` file.

File: randovania/games/samus_returns/gui/dialog/cosmetic_patches_dialog.py [hunk 1]
```diff
@@ -1,41 +1,111 @@
...
 
     @property
     def cosmetic_patches(self) -> MSRCosmeticPatches:
         return self._cosmetic_patches
 
-    def reset(self):
+    def _on_check_update(self, check: QtWidgets.QCheckBox, field_name: str, _: None) -> None:
+        self.preferences = dataclasses.replace(
+            self.preferences, **{field_name: check.isChecked()}  # type: ignore[arg-type]
+        )
```
Comment (ThanatosGit) [inline randovania/games/samus_returns/gui/dialog/cosmetic_patches_dialog.py:108]: Is this function even used? Can't see it on a quick glance.

File: test/games/samus_returns/layout/test_msr_cosmetic_patches.py
Comment (ThanatosGit) [inline test/games/samus_returns/layout/test_msr_cosmetic_patches.py:1]: I know the purpose is to exactly use a wrong type but try to use `# type: ignore` as comment above each line to tell mypy that it should ignore it at the following line.

File: test/games/samus_returns/gui/dialog/test_msr_cosmetic_patches_dialog.py
Comment (ThanatosGit) [inline test/games/samus_returns/gui/dialog/test_msr_cosmetic_patches_dialog.py:1]: It should be possible to just click every color picker button then the `_open_color_picker` function would be tested, too.
Like do `skip_qtbot.mouseClick(dialog.custom_laser_locked_color_button, QtCore.Qt.MouseButton.LeftButton)` for `test_custom_laser_color` etc. maybe?

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

File: randovania/games/samus_returns/gui/dialog/cosmetic_patches_dialog.py (hunk 1 above)
Comment (dyceron) [reply randovania/games/samus_returns/gui/dialog/cosmetic_patches_dialog.py:108]: Yeah it's definitely not used from what I can tell either. Just extra copy paste stuff

File: test/games/samus_returns/gui/dialog/test_msr_cosmetic_patches_dialog.py
Comment (dyceron) [reply test/games/samus_returns/gui/dialog/test_msr_cosmetic_patches_dialog.py:1]: It seems to work but creates an other dialog which is breaking the tests here. Maybe another left click could be added for confirming the choice, if that's possible to do.

Comment (ThanatosGit) [reply test/games/samus_returns/gui/dialog/test_msr_cosmetic_patches_dialog.py:1]: No idea on what we need to left click. I was under the impression that other games also don't test this, maybe that was the exact reason.
Then just leave it. We don't need coverage on everything and probably a waste of time here.

Comment (dyceron) [reply test/games/samus_returns/gui/dialog/test_msr_cosmetic_patches_dialog.py:1]: Then ill revert the latest changes

**Review by ThanatosGit — APPROVED** [review]

**Comment by dyceron** [comment]
Now supports displaying room names in game. Also fixes the default Grapple Laser Locked color.

image

**Review by Miepee — APPROVED** [review]

**Pull request merged**